File: go/problems/day2_test.go
package problems

import "testing"

var day2 Solvable[int] = Day2{}

func TestDay2_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day2_1")
	result := day2.Solve(input)

	expected := 8
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay2_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day2_2")
	result := day2.Solve(input)

	expected := 2331
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day2.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day2.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 2", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day2_1");
    const result = solve(input);
    assert.equal(result, 8);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day2_2");
    const result = solve(input);
    assert.equal(result, 2331);
  });
});
